stake-pool/cli/src/main.rs

```diff
@@ -374,66 +344,219 @@ fn command_create_pool(
+    Ok(())
+}
+
+fn setup_validator_list_account(
+    config: &Config,
+    validator_list_keypair: &Keypair,
+    validator_list_balance: u64,
+    validator_list_size: usize,
+) -> CommandResult {
+    let validator_list_account_info = config.rpc_client.get_account(&validator_list_keypair.pubkey());
+    if let Ok(account) = validator_list_account_info {
+        if account.owner == spl_stake_pool::id() {
+            if account.lamports == validator_list_balance {
+                println!("Validator list account {} already exists and is initialized", validator_list_keypair.pubkey());
+                return Ok(());
+            }
+        }
+    }
+
+    // Create validator list account
+    let instructions = vec![
         system_instruction::create_account(
             &config.fee_payer.pubkey(),
             &validator_list_keypair.pubkey(),
             validator_list_balance,
             validator_list_size as u64,
             &spl_stake_pool::id(),
         ),
-        // Account for the stake pool
-        system_instruction::create_account(
-            &config.fee_payer.pubkey(),
-            &stake_pool_keypair.pubkey(),
-            stake_pool_account_lamports,
-            get_packed_len::<StakePool>() as u64,
-            &spl_stake_pool::id(),
-        ),
-        // Initialize stake pool
+    ];
+
+    let signers = vec![config.fee_payer.as_ref(), validator_list_keypair];
+    let transaction = checked_transaction_with_signers(config, &instructions, &signers)?;
+
+    println!("Creating validator list account {}", validator_list_keypair.pubkey());
+    send_transaction(config, transaction)?;
+    Ok(())
+}
+
+fn initialize_stake_pool(
+    config: &Config,
+    stake_pool_keypair: &Keypair,
+    validator_list_keypair: &Keypair,
+    reserve_keypair: &Keypair,
+    mint_keypair: &Keypair,
+    pool_fee_account: &Pubkey,
+    deposit_authority: Option<Keypair>,
+    epoch_fee: Fee,
+    withdrawal_fee: Fee,
+    deposit_fee: Fee,
+    referral_fee: u8,
+    max_validators: u32,
+    withdraw_authority: &Pubkey,
+) -> CommandResult {
+    let stake_pool_account_info = config.rpc_client.get_account(&stake_pool_keypair.pubkey());
+
+    let stake_pool_account_lamports = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(get_packed_len::<StakePool>())?;
+        
+        
+    if let Ok(account) = stake_pool_account_info {
+        let stake_pool_manager = get_stake_pool(&config.rpc_client, &stake_pool_keypair.pubkey()).unwrap().manager;
+        if account.owner == spl_stake_pool::id() {
+            if stake_pool_manager == config.manager.pubkey() {
+                println!("Stake pool account {} already exists and is initialized", stake_pool_keypair.pubkey());
+                return Ok(());
+            } else {
+                println!("Stake pool account {} already exists but is not initialized", stake_pool_keypair.pubkey());
+            }
+        }
```

Comment:
This could unnecessarily panic -- rather than checking the manager, we can just check that the first byte in the data is equal to `AccountType::StakePool as u8` (if there is any data in the account) https://github.com/solana-labs/solana-program-library/blob/c4d51f20b8722d23b03e94c7075702fca52460dd/stake-pool/program/src/state.rs#L37